tiledb/sm/array/array.cc

```diff
@@ -2071,4 +2074,59 @@ void ensure_supported_schema_version_for_read(format_version_t version) {
   }
 }
 
+// NB: this is used to implement `tiledb_array_schema_get_enumeration_*`
+// but is defined here instead of array_schema to avoid a circular dependency
+// (array_directory depends on array_schema).
+void load_enumeration_into_schema(
+    Context& ctx, const std::string& enmr_name, ArraySchema& array_schema) {
+  if (array_schema.is_enumeration_loaded(enmr_name)) {
+    return;
+  }
+
+  auto tracker = ctx.resources().ephemeral_memory_tracker();
+
+  if (array_schema.array_uri().is_tiledb()) {
+    auto rest_client = ctx.resources().rest_client();
+    if (rest_client == nullptr) {
+      throw ArrayException(
+          "Error loading enumerations; Remote array schema with no REST "
+          "client.");
+    }
+
+    auto ret = rest_client->post_enumerations_from_rest(
+        array_schema.array_uri(),
+        array_schema.timestamp_start(),
+        array_schema.timestamp_end(),
+        ctx.resources().config(),
+        array_schema,
+        {enmr_name},
+        tracker);
+
+    // response is a map {schema: [enumerations]}
+    // we should be the only schema, and expect only one enumeration
```

Comment:
It'd be nice to add a REST test that evolves the array with an additional enumeration and then check if you get that new enumeration on the new array schema when you call `tiledb_array_schema_get_enumeration_from_name`.

Reply:
Sure, I'll add something in a follow-up PR

Reply:
https://github.com/TileDB-Inc/TileDB/pull/5368/commits/79c4eaf0ba9623e4c59e3f094f0b538093cb35f6